### PR TITLE
Improve main views

### LINKED ISSUE #161: Improve main views
Description: This issue involves enhancing the design of the primary views, including login, registration, and room list, to make them more visually appealing and user-friendly.

### PR DESCRIPTION
**Pull request by JonaQuaLabs**
fixes #161 

image

image

image

### CODE REVIEW

File: frontend/src/components/RoomsList.jsx [hunk 1]
```diff
@@ -3,22 +3,47 @@ import { Link as RouterLink } from 'react-router-dom';
...
+    background-color: #652ead;
+    color: #ffffff;
+  }
+  &&.custom-button:hover {
+    background-color: #391052;
+  }
+  &&.custom-button:disabled {
+    background-color: #cccccc;
+  }
+`;
```
Comment (marcovidonis) [inline frontend/src/components/RoomsList.jsx:29]: If we're going to change theme colours, I recommend [modifying the Material UI theme](https://mui.com/material-ui/customization/theming/) directly. This way, `Button` elements will still work with the `color='primary'` prop, etc. And we don't need to set custom styles in multiple places, which can soon cause issues with maintainability.

Also, the custom palette will be applied everywhere (the nav bar was not updated to the new purple theme colour), and will work better if we enable dark mode. MUI will also generate lighter and darker versions, as well as a contrast colour for text.

File: frontend/src/components/RoomsList.jsx [hunk 2]
```diff
@@ -3,22 +3,47 @@ import { Link as RouterLink } from 'react-router-dom';
...
+
+const StyledCard = styled(Card)`
+  &&.custom-card {
+    border: 1px solid #f0e4ff;
```
Comment (marcovidonis) [inline frontend/src/components/RoomsList.jsx:33]: When styling MUI components, also consider having a look at [the `sx` prop](https://mui.com/material-ui/react-box/#the-sx-prop), which has access to the theme palette, among other things. This saves you from having to hunt down wherever you set custom colours, if you want to make changes to the theme later.

File: frontend/src/views/SignIn.jsx [hunk 3]
```diff
@@ -1,24 +1,75 @@
...
+    background-color: #652ead;
+    color: #ffffff;
+  }
+  &&.custom-button:hover {
+    background-color: #391052;
+  }
+  &&.custom-button:disabled {
+    background-color: #cccccc;
+  }
+`;
```
Comment (marcovidonis) [inline frontend/src/views/SignIn.jsx]: See comment above about theming.

File: frontend/src/views/SignIn.jsx [hunk 4]
```diff
@@ -1,24 +1,75 @@
...
+
+const StyledLink = styled(Link)`
+  color: #007bff;
```
Comment (marcovidonis) [inline frontend/src/views/SignIn.jsx]: See above.

File: frontend/src/views/SignIn.jsx [hunk 5]
```diff
@@ -49,71 +101,93 @@ function SignIn() {
...
-            {errors.email?.message}
-          </Typography>
           <TextField
             margin="normal"
-            required
             fullWidth
             name="password"
             label="Password"
-            type="password"
+            type={showPassword ? 'text' : 'password'}
```
Comment (marcovidonis) [inline frontend/src/views/SignIn.jsx:113]: Nice! 👍

File: frontend/src/views/SignUp.jsx [hunk 6]
```diff
@@ -1,24 +1,78 @@
...
+    background-color: #652ead;
+    color: #ffffff;
+  }
+  &&.custom-button:hover {
+    background-color: #391052;
+  }
+  &&.custom-button:disabled {
+    background-color: #cccccc;
+  }
+`;
```
Comment (marcovidonis) [inline frontend/src/views/SignUp.jsx]: See above.

File: frontend/src/views/SignUp.jsx [hunk 7]
```diff
@@ -1,24 +1,78 @@
...
+
+const StyledLink = styled(Link)`
+  color: #007bff;
```
Comment (marcovidonis) [inline frontend/src/views/SignUp.jsx]: See above.

File: frontend/src/views/SignUp.jsx [hunk 8]
```diff
@@ -1,24 +1,78 @@
...
+`;
+
+const StyledHeader = styled(Typography)`
+  font-weight: bold;
+  text-align: center;
+`;
+
+const StyledAvatar = styled(Avatar)`
+  background-color: #dbbcd7;
```
Comment (marcovidonis) [inline frontend/src/views/SignUp.jsx]: See above.

**Review by marcovidonis — COMMENTED** [review]
I recommend making use of the theme customisation capabilities in Material UI, which will make the code more maintainable (and if we change MUI for something else in the future, we'll still want all the styling in one place).

While we're updating the UI, I think the app name in the navbar could use a better font (or a logo). Also, only the text should be a link, while now the width of the element is almost the whole navbar. And we should remove the hover effect.

**Comment by JonaQuaLabs** [comment]
@marcovidonis First of all, thank you for your comments, below is a brief description of what was improved

This new pull request involves moving component styles to a separate file in order to create more generic styles for components. This approach allows for easier styling modifications and ensures that styles can be reused across multiple components.
We believe this is the correct approach for improving our styling architecture and will lead to better maintainability and scalability of our application. However, we are open to any feedback or suggestions.

FYI Please note that while we have made significant progress with this pull request, there are still a few modifications that need to be made. Due to time constraints, we will address these modifications in upcoming sprints.

**Review by yulianag-qualabs — APPROVED** [review]

**Review by fpyoung — APPROVED** [review]

**Pull request merged**